Interface version canonicalization (#536)
WebAssembly/component-model: design/mvp/Binary.md

  `(result (own $R))`, where `$R` is the resource labeled `r`.
* Validation of `[method]` names requires the first parameter of the function
  to be `(param "self" (borrow $R))`, where `$R` is the resource labeled `r`.

> * `<valid semver>` is as defined by [https://semver.org](https://semver.org/)

Author: `<valid semver>` wasn't referenced in this file.

>   with a `canonversion` and that the concatenation of the `canonversion` and the

Reviewer: ```suggestion
  matches `canonversion` and that the concatenation of the `canonversion` and the
```
